src/rest_vol_dataset.c

```diff
@@ -394,7 +405,23 @@ RV_dataset_open(void *obj, const H5VL_loc_params_t *loc_params, const char *name
 #endif
 
     /* Set up a Dataspace for the opened Dataset */
-    if ((dataset->u.dataset.space_id = RV_parse_dataspace(response_buffer.buffer)) < 0)
+
+    /* If this is another view of an already-opened dataset, make them share the same dataspace
+     * so that changes to it (e.g. resizes) are visible to both views */
+    while (rv_hash_table_iter_has_more(&iterator)) {
+        other_dataset = (RV_object_t *)rv_hash_table_iter_next(&iterator);
```

Comment:
This is fine for now functionality wise, but we need a better solution for this rather than iterating through the hash table on each dataset open. I'd suggest refactoring this whole PR later where rather than inserting each object pointer into a hash table for the ID type, we hash the object's URI and insert that hashed value into the hash table for the ID type. Then, on each dataset (of other object) open, you only need to hash the URI here and do a single lookup to see if an object with this URI is already open.